lib/queuePopulator/IngestionPopulator.js

```diff
@@ -17,6 +18,9 @@ const {
     zkStateProperties,
 } = require('../../extensions/ingestion/constants');
 
+const POLL_INTERVAL_MS = 50;
+const MAX_PARALLEL_READERS = 5;
```

Comment:
Since this seems to be a sensitive configuration value, I suggest to expose it via the config so that it can be tuned if need be.